### PR TITLE
Fix Flooring in /statcalc

### PR DESCRIPTION
**Pull request by PartMan7**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-8607763

I've used ~~ instead of Math.floor for performance reasons, but it shouldn't be much of a difference either way.

### CODE REVIEW

**Comment by urkerab** [comment]
If you were worried about performance then wouldn't you use `(ev >> 2)` in the first place?

**Comment by PartMan7** [comment]
I'll do that, then. Didn't know about it. 'o.o

**Comment by monsanto** [comment]
I think optimizations which make the code harder to read should be accompanied by a benchmark proving the gain is worth it.

File: server/chat-commands/info.ts [hunk 1]
```diff
@@ -1411,15 +1411,15 @@ export const commands: ChatCommands = {
...
+					baseStat = Math.ceil((100 * realStat - 10 - level * (Math.floor(ev / 4) + iv + 100)) / (2 * level));
 				} else {
 					if (!positiveMod) {
 						realStat *= (2 + modifier) / 2;
 					} else {
 						realStat *= 2 / (2 + modifier);
 					}
 
-					baseStat = Math.ceil((100 * Math.ceil(realStat) - nature * (level * (ev / 4 + iv) + 500)) / (2 * level * nature));
+					baseStat = Math.ceil((100 * Math.ceil(realStat) - nature * (level * (Math.floor(ev / 4) + iv) + 500)) / (2 * level * nature));
```
Comment (Zarel) [inline server/chat-commands/info.ts:1422]:
```suggestion
					baseStat = Math.ceil(
						(100 * Math.ceil(realStat) - nature * (level * (Math.floor(ev / 4) + iv) + 500)) /
						(2 * level * nature)
					);
```

This is probably the most readable way to come in under the line limit.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**